### PR TITLE
Load a game on Resume and F9 to quickload in menu

### PR DESCRIPTION
**Pull request by bsxf-47**
Implements both features in Feature request #45. 

I'm not entirely sure if one doesn't need to do other operations before loading a game using ARXMenu_LoadQuest() and ARX_Quickload(), so if anything's wrong, just let me know and I'll modify it accordingly.

### CODE REVIEW

**Comment by bsxf-47** [comment]
Changed the second commit to not allow quicksaving in the menu and grouped conditions under one AND statement.

File: src/gui/MainMenu.cpp [hunk 1]
```diff
@@ -1777,7 +1777,12 @@ void MainMenu::init()
 
 void MainMenu::onClickedResumeGame(){
 	pTextManage->Clear();
-	ARXMenu_ResumeGame();
+	if(!g_canResumeGame) {
+		//the most recent savegame is at the beginning of the list
+		ARXMenu_LoadQuest(SavegameHandle(0));
```
Comment (dscharrer) [inline src/gui/MainMenu.cpp]: Couldn't this be ARX_QuickLoad()? - you use that for F9 from the menu.

File: src/gui/MainMenu.cpp [hunk 2]
```diff
@@ -1798,8 +1803,10 @@ MENUSTATE MainMenu::Update() {
 			m_resumeGame->SetCheckOn();
 			m_resumeGame->lColor = Color(232, 204, 142);
 		} else {
-			m_resumeGame->SetCheckOff();
-			m_resumeGame->lColor = Color(127, 127, 127);
+			if(savegames.size() == 0) {
```
Comment (dscharrer) [inline src/gui/MainMenu.cpp]: Use `else if` - no need for additional nesting.

File: src/core/ArxGame.cpp [hunk 3]
```diff
@@ -1312,14 +1312,14 @@ void ArxGame::doFrame() {
 	if(cinematicIsStopped()
 	   && !cinematicBorder.isActive()
 	   && !BLOCK_PLAYER_CONTROLS
-	   && ARXmenu.currentmode == AMCM_OFF
 	) {
 		
-		if(GInput->actionNowPressed(CONTROLS_CUST_QUICKLOAD)) {
+		if(GInput->actionNowPressed(CONTROLS_CUST_QUICKLOAD) && (savegames.size() > 0)) {
```
Comment (dscharrer) [inline src/core/ArxGame.cpp]: No need for the extra parentheses around `savegames.size() > 0`.

File: src/core/ArxGame.cpp [hunk 4]
```diff
@@ -1312,14 +1312,14 @@ void ArxGame::doFrame() {
...
+			ARXmenu.currentmode = AMCM_OFF;
 			ARX_QuickLoad();
 		}
 		
-		if(GInput->actionNowPressed(CONTROLS_CUST_QUICKSAVE)) {
+		if(GInput->actionNowPressed(CONTROLS_CUST_QUICKSAVE) && (ARXmenu.currentmode == AMCM_OFF)) {
```
Comment (dscharrer) [inline src/core/ArxGame.cpp]: No need for the extra parentheses around `ARXmenu.currentmode == AMCM_OFF`.

**Comment by bsxf-47** [comment]
Thanks for the feedback. I updated the commits with the suggested changes.

**Comment by dscharrer** [comment]
Merged as commits 496d02f...cda2577.

**Pull request closed without merging**